File: ahorcado.py
import random

def escoger_palabra():
    palabras = ['python', 'ingenieria', 'sistemas', 'laptop', 'autonomo', 'programacion']
    return random.choice(palabras)

def mostrar_progreso(palabra, letras_adivinadas):
    progreso = [letra if letra in letras_adivinadas else '_' for letra in palabra]
    return ' '.join(progreso)

def jugar_ahorcado():
    palabra = escoger_palabra()
    letras_adivinadas = []
    intentos_restantes = 6
    juego_terminado = False

    print("¡Bienvenido al juego del ahorcado!")

    while not juego_terminado:
        print(f"\nPalabra: {mostrar_progreso(palabra, letras_adivinadas)}")
        print(f"Te quedan {intentos_restantes} intentos.")
        
        intento = input("Adivina una letra: ").lower()

        if len(intento) != 1 or not intento.isalpha():
            print("Por favor, introduce solo una letra.")
            continue

        if intento in letras_adivinadas:
            print(f"Ya adivinaste la letra {intento}. Intenta otra.")
            continue

        letras_adivinadas.append(intento)

        if intento in palabra:
            print(f"¡Bien! La letra {intento} está en la palabra.")
        else:
            intentos_restantes -= 1
            print(f"Lo siento, la letra {intento} no está en la palabra.")
        
        if all(letra in letras_adivinadas for letra in palabra):
            print(f"¡Felicidades! Adivinaste la palabra: {palabra}")
            juego_terminado = True
        elif intentos_restantes == 0:
            print(f"¡Ahorcado! La palabra era: {palabra}")
            juego_terminado = True

jugar_ahorcado()

File: ahorcado.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cstdlib>
#include <ctime>

using namespace std;

void displayWord(const string &word, const vector<bool> &guessed) {
    for (int i = 0; i < word.size(); i++) {
        if (guessed[i]) {
            cout << word[i] << " ";
        } else {
            cout << "_ ";
        }
    }
    cout << endl;
}

bool isWordGuessed(const vector<bool> &guessed) {
    for (bool letterGuessed : guessed) {
        if (!letterGuessed) {
            return false;
        }
    }
    return true;
}

void drawHangman(int attemptsLeft) {
    switch (attemptsLeft) {
        case 6:
            cout << "_________" << endl;
            cout << "|       |" << endl;
            cout << "|" << endl;
            cout << "|" << endl;
            cout << "|" << endl;
            cout << "|" << endl;
            cout << "|" << endl;
            break;
        case 5:
            cout << "_________" << endl;
            cout << "|       |" << endl;
            cout << "|       O" << endl;
            cout << "|" << endl;
            cout << "|" << endl;
            cout << "|" << endl;
            cout << "|" << endl;
            break;
        case 4:
            cout << "_________" << endl;
            cout << "|       |" << endl;
            cout << "|       O" << endl;
            cout << "|       |" << endl;
            cout << "|" << endl;
            cout << "|" << endl;
            cout << "|" << endl;
            break;
        case 3:
            cout << "_________" << endl;
            cout << "|       |" << endl;
            cout << "|       O" << endl;
            cout << "|      /|" << endl;
            cout << "|" << endl;
            cout << "|" << endl;
            cout << "|" << endl;
            break;
        case 2:
            cout << "_________" << endl;
            cout << "|       |" << endl;
            cout << "|       O" << endl;
            cout << "|      /|\\" << endl;
            cout << "|" << endl;
            cout << "|" << endl;
            cout << "|" << endl;
            break;
        case 1:
            cout << "_________" << endl;
            cout << "|       |" << endl;
            cout << "|       O" << endl;
            cout << "|      /|\\" << endl;
            cout << "|      /" << endl;
            cout << "|" << endl;
            cout << "|" << endl;
            break;
        case 0:
            cout << "_________" << endl;
            cout << "|       |" << endl;
            cout << "|       O" << endl;
            cout << "|      /|\\" << endl;
            cout << "|      / \\" << endl;
            cout << "|" << endl;
            cout << "|" << endl;
            break;
    }
}

int main() {
    srand(static_cast<unsigned int>(time(0))); // Semilla para números aleatorios

    // Lista de palabras para adivinar
    vector<string> words = {"computadora", "programacion", "ahorcado", "algoritmo", "sistemas"};
    string word = words[rand() % words.size()]; // Selecciona una palabra aleatoria

    vector<bool> guessed(word.size(), false);   // Letras adivinadas
    int attemptsLeft = 6;                       // Intentos restantes
    vector<char> incorrectGuesses;              // Letras incorrectas

    cout << "¡Bienvenido al juego de Ahorcado!" << endl;

    while (attemptsLeft > 0 && !isWordGuessed(guessed)) {
        drawHangman(attemptsLeft);
        displayWord(word, guessed);
        cout << "Letras incorrectas: ";
        for (char c : incorrectGuesses) {
            cout << c << " ";
        }
        cout << endl;

        cout << "Intentos restantes: " << attemptsLeft << endl;
        cout << "Introduce una letra: ";
        char guess;
        cin >> guess;

        bool isCorrectGuess = false;
        for (int i = 0; i < word.size(); i++) {
            if (word[i] == guess && !guessed[i]) {
                guessed[i] = true;
                isCorrectGuess = true;
            }
        }

        if (!isCorrectGuess) {
            incorrectGuesses.push_back(guess);
            attemptsLeft--;
        }
    }

    if (isWordGuessed(guessed)) {
        cout << "¡Felicidades! Adivinaste la palabra: " << word << endl;
    } else {
        drawHangman(0);
        cout << "Lo siento, has perdido. La palabra era: " << word << endl;
    }

    return 0;
}
